src/java/org/apache/cassandra/service/snapshot/SnapshotManager.java

```diff
@@ -116,16 +109,24 @@ public synchronized void addSnapshot(TableSnapshot snapshot)
     @VisibleForTesting
     protected synchronized void loadSnapshots()
     {
-        logger.debug("Loading snapshots");
-        snapshotLoader.get().forEach(this::addSnapshot);
+        SnapshotLoader loader = new SnapshotLoader(DatabaseDescriptor.getAllDataFileLocations());
+        addSnapshots(loader.loadSnapshots());
+    }
+
+    @VisibleForTesting
+    protected synchronized void addSnapshots(Collection<TableSnapshot> snapshots)
+    {
+        logger.debug("Adding snapshots: {}.", Joiner.on(", ").join(snapshots.stream().map(s -> s.getId()).collect(Collectors.toList())));
```

Comment:
this looks better but what I see is that we have "keyspace:table:id" rather than _name of that snapshot_ (snapshot tag).

Reply:
Ahaaaa I get this, so after "id" there is the snapshot tag, the whole name has four components, ks:tb:id:tag. I overlooked the last part. All good.